kogito-codegen-modules/kogito-codegen-predictions/src/main/java/org/kie/kogito/codegen/prediction/PMMLRestResourceGenerator.java

```diff
@@ -106,17 +122,107 @@ void setPathValue(ClassOrInterfaceDeclaration template) {
+                .getAnnotations();
+        switch(context.name()) {
+            case "Quarkus":
+                setQuarkusOASAnnotations(annotations, inputRef, outputRef);
+                break;
+            case "Spring":
+                setSpringOASAnnotations(annotations, inputRef, outputRef);
+                break;
+            default:
+                // noop
+        }
     }
 
-    private void initializeApplicationField(FieldDeclaration fd) {
+    void setDescriptiveOASAnnotations(ClassOrInterfaceDeclaration template, String jsonFile, String inputRef) {
+        String outputRef = String.format("/%s#/definitions/OutputSet", jsonFile);
+        NodeList<AnnotationExpr> annotations = template.getMethodsByName("descriptive").get(0)
+                .getAnnotations();
+        switch(context.name()) {
+            case "Quarkus":
+                setQuarkusOASAnnotations(annotations, inputRef, outputRef);
+                break;
+            case "Spring":
+                setSpringOASAnnotations(annotations, inputRef, outputRef);
+                break;
+            default:
+                // noop
+        }
+    }
+
+    void setQuarkusOASAnnotations(NodeList<AnnotationExpr> annotations, String inputRef, String outputRef) {
+        Optional<MemberValuePair> ref = getRefMemberValuePair(annotations, QUARKUS_REQUEST_BODY, QUARKUS_SCHEMA);
+        ref.ifPresent(rf -> rf.setValue(new StringLiteralExpr(inputRef)));
+        ref = getRefMemberValuePair(annotations, QUARKUS_API_RESPONSE, QUARKUS_SCHEMA);
+        ref.ifPresent(rf -> rf.setValue(new StringLiteralExpr(outputRef)));
+    }
+
+    void setSpringOASAnnotations(NodeList<AnnotationExpr> annotations, String inputRef, String outputRef) {
+        Optional<MemberValuePair> ref = getRefMemberValuePair(annotations, SPRING_REQUEST_BODY, SPRING_SCHEMA);
+        ref.ifPresent(rf -> rf.setValue(new StringLiteralExpr(inputRef)));
+        ref = getRefMemberValuePair(annotations, SPRING_API_RESPONSE, SPRING_SCHEMA);
+        ref.ifPresent(rf -> rf.setValue(new StringLiteralExpr(outputRef)));
+    }
+
+    void initializeApplicationField(FieldDeclaration fd) {
         fd.getVariable(0).setInitializer(new ObjectCreationExpr().setType(appCanonicalName));
     }
+
+    Optional<MemberValuePair> getRefMemberValuePair(NodeList<AnnotationExpr> source, String annotationName, String schemaName) {
+        return getAnnotationExpr(source, annotationName)
+                .map(annExpr -> getMemberValuePairFromAnnotation(annExpr, CONTENT))
+                .filter(Optional::isPresent)
+                .map(content -> getMemberValuePairFromMemberValuePair(content.get(), SCHEMA))
+                .filter(Optional::isPresent)
+                .map(schema -> getNormalAnnotationExprFromMemberValuePair(schema.get(), schemaName))
+                .filter(Optional::isPresent).flatMap(schemaAnnotation -> getMemberValuePairFromAnnotation(schemaAnnotation.get(), REF));
```

Comment:
You can just apply `map` without the need of a filter
```suggestion
        return getAnnotationExpr(source, annotationName)
                .map(annExpr -> getMemberValuePairFromAnnotation(annExpr, CONTENT))
                .map(content -> getMemberValuePairFromMemberValuePair(content.get(), SCHEMA))
                .map(schema -> getNormalAnnotationExprFromMemberValuePair(schema.get(), schemaName))
                .flatMap(schemaAnnotation -> getMemberValuePairFromAnnotation(schemaAnnotation.get(), REF));
```

Reply:
@danielezonca 
without the filter, IDE warns about "Optional.get without 'isPresent' check" - 
Maybe is a different expression you are thinking of - that I would be glad to use

Reply:
You are right, you need to use `flatMap` instead of `map`
```suggestion
        return getAnnotationExpr(source, annotationName)
                .flatMap(annExpr -> getMemberValuePairFromAnnotation(annExpr, CONTENT))
                .flatMap(content -> getMemberValuePairFromMemberValuePair(content, SCHEMA))
                .flatMap(schema -> getNormalAnnotationExprFromMemberValuePair(schema, schemaName))
                .flatMap(schemaAnnotation -> getMemberValuePairFromAnnotation(schemaAnnotation, REF));
```